### PR TITLE
client & engine client

### PR DESCRIPTION
**Pull request by tanner0101**
- [x] engineclient
- [x] client protocol
- [x] support http/2 (@Joannis do we have cleartext support yet? how do we go about dynamically upgrading?)

### CODE REVIEW

File: Sources/TCP/Streams/Client.swift [hunk 1]
```diff
@@ -238,7 +238,8 @@ public final class TCPClient: Async.Stream, ClosableStream {
         // for some reason you can't cancel a suspended write source
         // if you remove this line, your life will be ruined forever!!!
         if self.inputBuffer.count == 0 {
-            writeSource?.resume()
+            /// FIXME: this crashes for client?
```
Comment (tanner0101) [inline Sources/TCP/Streams/Client.swift]: @Joannis when we take a look at cleaning up the Writeable callback, we should also look into what is going on here.

**Comment by Joannis** [comment]
@tanner0101 Cleartext support for HTTP/2 is not implemented on purpose. Almost the entire internet unanimously agreed not to support it in an implementation. None of the major web browsers and (almost) none of the web servers I've seen accept cleartext HTTP/2 it seems.

**Comment by Joannis** [comment]
HTTP/2 over SSL needs support from the start of the SSL handshake. So you don't need a special implementation for upgrading.

File: Sources/Fluent/Query/Filter/FilterMethod.swift [hunk 2]
```diff
@@ -26,4 +26,23 @@ extension QueryBuilder {
         let filter = QueryFilter(entity: Model.entity, method: value)
         return addFilter(filter)
     }
+
+    @discardableResult
+    public func filter(
```
Comment (Joannis) [inline Sources/Fluent/Query/Filter/FilterMethod.swift]: inline docs are missing

File: Sources/HTTP/Streams/Server.swift [hunk 3]
```diff
@@ -33,9 +40,41 @@ public final class HTTPServer: Async.OutputStream, ClosableStream {
...
+
     /// See ClosableStream.close
     public func close() {
         socket.close()
         outputStream.close()
     }
 }
+
+/// Capable of being started by an HTTP server.
+public protocol HTTPStartable {
```
Comment (Joannis) [inline Sources/HTTP/Streams/Server.swift]: We can remove this protocol I think.

**Pull request merged**